src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -91,11 +95,26 @@ public LowLevelClientMethod BuildOperationMethodChain()
             return new LowLevelClientMethod(protocolMethodSignature, convenienceMethod, _restClientMethod, requestBodyType, responseBodyType, diagnostic, _protocolMethodPaging, Operation.LongRunning, _conditionHeaderFlag);
         }
 
-        private bool ShouldConvenienceMethodGenerated(ReturnTypeChain returnTypeChain)
+        private bool ShouldConvenienceMethodGenerated()
         {
             return Operation.GenerateConvenienceMethod
                 && (_orderedParameters.Where(parameter => parameter.Convenience != KnownParameters.CancellationTokenParameter).Any(parameter => !IsParameterTypeSame(parameter.Convenience, parameter.Protocol))
-                || !returnTypeChain.Convenience.Equals(returnTypeChain.Protocol));
+                || !_returnType.Convenience.Equals(_returnType.Protocol));
+        }
+
+        private bool ShouldOverloadConvenienceMethod()
+        {
+            if (!ShouldConvenienceMethodGenerated() || _sourceInputModel == null)
+            {
+                return false;
+            }
+            var existingMethod = _sourceInputModel.FindForMethod(Operation.Name.ToCleanName());
```

Comment:
Do we have a concrete way to get the name of this method? We do not need to calculate that again here

Reply:
You mean cache the clean name in operation, or?

Reply:
maybe, having a centralized place for the name which this generated method is meant to have.
For instance, since this class is a builder for an operation, we should have a place that we return the operation with a proper name. We can extract that name and put it into at least a variable